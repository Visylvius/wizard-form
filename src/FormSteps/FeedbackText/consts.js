const FEEDBACK_TEXT = 'feedbackText';

export { FEEDBACK_TEXT };
