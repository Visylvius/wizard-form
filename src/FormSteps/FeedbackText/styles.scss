@import '../../variables';

.FeedbackText-characterLimit {
  display: flex;
  color: $character-limit;
  font-size: 10px;
  justify-content: flex-end;
}