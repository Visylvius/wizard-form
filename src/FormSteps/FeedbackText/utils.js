const calculateRemainingChars = totalCount => {
  return 1000 - totalCount;
};

export { calculateRemainingChars };
