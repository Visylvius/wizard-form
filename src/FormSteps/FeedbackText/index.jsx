import React from 'react';

import TextArea from '../../Components/TextArea';

import { calculateRemainingChars } from './utils';
import { FEEDBACK_TEXT } from './consts';
import './styles.scss';

const FeedbackText = ({ currentStep, testId, onChange, feedbackValue }) => {
  if (currentStep !== 5) {
    return null;
  }
  return (
    <div>
      <TextArea
        data-testid={testId}
        onChange={onChange}
        value={feedbackValue}
        maxLength={1000}
        hasLabel
        label='Please share your feedback.'
        name={FEEDBACK_TEXT}
        placeholder='your answer'
      />
      <div className='FeedbackText-characterLimit'>{`${calculateRemainingChars(
        feedbackValue.length,
      )} characters remaining`}</div>
    </div>
  );
};

export default FeedbackText;
