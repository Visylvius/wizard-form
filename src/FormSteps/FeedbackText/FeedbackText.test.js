import React from 'react';
import renderer from 'react-test-renderer';
import { render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import FeedbackText from './index';

describe('FeedbackText (Unit)', () => {
  const mockOnChange = jest.fn();
  const renderFeedbackText = props =>
    render(
      <FeedbackText
        currentStep={5}
        feedbackValue=''
        onChange={mockOnChange}
        testId='FormStep-TextArea'
      />,
    );

  describe('render', () => {
    test('should render properly', () => {
      const tree = renderer.create(
        <FeedbackText
          currentStep={5}
          feedbackValue='feedbackVal'
          onChange={jest.fn}
        />,
      );

      expect(tree).toMatchSnapshot();
    });

    test('should render null if the step is not 5', () => {
      const tree = renderer.create(
        <FeedbackText
          currentStep={3}
          feedbackValue='feedbackVal'
          onChange={jest.fn}
        />,
      );

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Intergration', () => {
    test('should properly call onChange', async () => {
      const { getByTestId } = renderFeedbackText();

      userEvent.type(getByTestId('FormStep-TextArea'), 'testing 123');

      await waitFor(() => expect(mockOnChange).toHaveBeenCalled());
    });
  });
});
