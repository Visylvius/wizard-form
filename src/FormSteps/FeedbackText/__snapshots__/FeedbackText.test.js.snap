// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackText (Unit) render should render null if the step is not 5 1`] = `null`;

exports[`FeedbackText (Unit) render should render properly 1`] = `
<div>
  <label
    className="TextArea-label"
  >
    Please share your feedback.
  </label>
  <textarea
    className="TextArea-component"
    maxLength={1000}
    name="feedbackText"
    onChange={[Function]}
    placeholder="your answer"
    value="feedbackVal"
  />
  <div
    className="FeedbackText-characterLimit"
  >
    989 characters remaining
  </div>
</div>
`;
